File: Tipper/app/src/main/java/com/example/tipper/inmemory/BMIResultData.java
package com.example.tipper.inmemory;

import com.example.tipper.BMIActivity;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Author Michał Wadas s20495
 */
public class BMIResultData {
    private final List<BMIActivity.BMIResult> resultBMI = new ArrayList<>();
    private static BMIResultData bmiResultData;

    private BMIResultData() {}

    public static BMIResultData getBmiResultData() {
        if (bmiResultData == null) {
            bmiResultData = new BMIResultData();
        }
        return bmiResultData;
    }

    public void addBMIResult(double bmi) {
        BMIActivity.BMIResult result = new BMIActivity.BMIResult(bmi, new Date());
        resultBMI.add(result);
    }
    public List<BMIActivity.BMIResult> getBMIResults() {
        return resultBMI;
    }


}

File: Tipper/app/src/main/java/com/example/tipper/inmemory/BMIResultData.kt
package com.example.tipper.inmemory

import com.example.tipper.BMIActivity.BMIResult
import java.util.*

/**
 * Author Michał Wadas s20495
 */
class BMIResultData private constructor() {
    private val resultBMI: MutableList<BMIResult> = ArrayList()
    fun addBMIResult(bmi: Double) {
        val result = BMIResult(bmi, Date())
        resultBMI.add(result)
    }

    val bMIResults: List<BMIResult>
        get() = resultBMI

    companion object {
        @kotlin.jvm.JvmStatic
        var bmiResultData: BMIResultData? = null
            get() {
                if (field == null) {
                    field = BMIResultData()
                }
                return field
            }
            private set
    }
}